driver/plan_manager.py

```diff
@@ -1,28 +1,24 @@
 import itertools
-import os
-import os.path
+from pathlib import Path
 import re
 
 from . import returncodes
 
 
-_PLAN_INFO_REGEX = re.compile(r"; cost = (\d+) \((unit cost|general cost)\)\n")
+_PLAN_INFO_REGEX = re.compile(r"; cost = (\d+) \((unit cost|general cost)\)")
 
 
-def _read_last_line(filename):
-    line = None
-    with open(filename) as input_file:
-        for line in input_file:
-            pass
-    return line
+def _read_last_line(path: Path):
+    lines = path.read_text().splitlines()
+    if lines:
+        return lines[-1]
```

Comment:
This is not a good change. The old algorithm stored O(1) lines of the input file in memory (plus the amount of buffering that is used, but this is a constant), the new code stores all of it in memory.

While plans are *usually* not long, I've seen plans with tens of millions of steps, and the old O(1) code was working fine.